demos/sample_azure_iot_adu/sample_azure_iot_adu.c

```diff
@@ -372,16 +372,38 @@ static void prvConnectHTTP( AzureIoTTransportInterface_t * pxHTTPTransport,
 }
 
 /**
- * @brief This is a hack. TODO: Replace with a proper url-parsing api.
+ * @brief Parses the full ADU file URL into a host (FQDN) and its path.
+ *
+ * @param xFileUrl ADU file Url to be parsed.
+ * @param pucBuffer Buffer to be used for pxHost and pxPath.
+ * @param ulBufferSize Size of pucBuffer
+ * @param pxHost Where the host part of the url is stored, including a null terminator.
+ *               The size of this span is equal to the length of the host address plus the
+ *               size of the null-terminator char.
+ * @param pxPath Where the path part of the url is stored.
  */
-static void prvParseAduUrl( az_span xUrl,
-                            az_span * pxHost,
-                            az_span * pxPath )
+static void prvParseAduFileUrl( AzureIoTADUUpdateManifestFileUrl_t xFileUrl,
```

Comment:
for the final product we can't have `az_span` things leaking out to user code.

Reply:
That shall be fixed. Thanks.

Reply:
Done.